### PR TITLE
[#205] Add zstd compression to the binary cache

### LINKED ISSUE #205: Support brotli/zstd compression besides xz
In https://github.com/cachix/cachix/issues/55 it was pointed out that zstd/brotli both [can achieve much faster transfer speeds](https://community.centminmod.com/threads/round-3-compression-comparison-benchmarks-zstd-vs-brotli-vs-pigz-vs-bzip2-vs-xz-etc.17259/).

### PR DESCRIPTION
**Pull request by Player-205**
Problem: xz don't have sufficient transfer speed of binary cache

Solution: add zstd as another compression option to the push strategy

Fixes #205

### CODE REVIEW

File: cachix/src/Cachix/Client/Commands.hs [hunk 1]
```diff
@@ -224,11 +227,22 @@ getPushParams env pushOpts name = do
   pushSecret <- findPushSecret (config env) name
   store <- wait (storeAsync env)
   authToken <- Config.getAuthTokenMaybe (config env)
+  compressionMode <- case authToken of
```
Comment (domenkozar) [inline cachix/src/Cachix/Client/Commands.hs:230]: This should rather case on `PushSecret` and disable this mode for when signing key is used.

Comment (Player-205) [reply cachix/src/Cachix/Client/Commands.hs:230]: What the difference between auth token and push token in my case? AFAIU, I should just check PushSecret constructor and ignore its fields, right?

Comment (domenkozar) [reply cachix/src/Cachix/Client/Commands.hs:230]: Almost zero, but you can have auth token and signing key, so it's best to look into `PushSecret` to make sure we cover both cases correctly. No need to look at the fields.

**Review by domenkozar — APPROVED** [review]
Could you squash please?

**Comment by domenkozar** [comment]
Continuing in #472, thanks!

**Pull request closed without merging**